script.js

```diff
@@ -1,4 +1,593 @@
+//FUNCTION 1 - Create deck.
+var makeDeck = function () {
+  var cardDeck = [];
+  var suits = ["hearts", "diamonds", "clubs", "spades"];
+  var emoji = ["♥️", "♦️", "♣️", "♠️"];
+
+  var suitIndex = 0;
+  while (suitIndex < suits.length) {
+    var currentSuit = suits[suitIndex];
+    var currentEmoji = emoji[suitIndex];
+
+    var cardCounter = 1;
+    while (cardCounter <= 13) {
+      // By default, the card name is the same as cardCounter
+      // By default, card rank is the same as cardCounter.
+      var cardName = cardCounter;
+      var cardRank = cardCounter;
+
+      // If counter is 1, 11, 12, or 13, set cardName to the ace or face card's name
+      // Also set card rank for ace to 11, jack/queen/king to 10.
+      // Let default card rank for ace to be 11, so that we can identify if anyone has ace later by sorting array, and also if score > 21 when ace is 11, we can allow ace to be 1 so that the score can be decreased below 21.
+      if (cardName == 1) {
+        cardName = "Ace";
+        cardRank = 11;
+      } else if (cardName == 11) {
+        cardName = "Jack";
+        cardRank = 10;
+      } else if (cardName == 12) {
+        cardName = "Queen";
+        cardRank = 10;
+      } else if (cardName == 13) {
+        cardName = "King";
+        cardRank = 10;
+      }
+
+      // Create a new card with the current name, suit, emoji and rank
+      var card = {
+        name: cardName,
+        suit: currentSuit,
+        rank: cardRank,
+        emoji: currentEmoji,
+      };
+
+      // Add the new card to the deck
+      cardDeck.push(card);
+      cardCounter += 1;
+    }
+    suitIndex += 1;
+  }
+  return cardDeck;
+};
+
+//FUNCTION 2 - Shuffle deck.
+// Get a random index ranging from 0 (inclusive) to max (exclusive).
+var getRandomIndex = function (max) {
+  return Math.floor(Math.random() * max);
+};
+
+// Shuffle the elements in the cardDeck array
+var shuffleCards = function (cardDeck) {
+  var currentIndex = 0;
+  while (currentIndex < cardDeck.length) {
+    var randomIndex = getRandomIndex(cardDeck.length);
+    var randomCard = cardDeck[randomIndex];
+    var currentCard = cardDeck[currentIndex];
+    cardDeck[currentIndex] = randomCard;
+    cardDeck[randomIndex] = currentCard;
+    currentIndex = currentIndex + 1;
+  }
+  return cardDeck;
+};
+
+//FUNCTION 3a - DRAW HAND. i.e. Pop 2 initial cards from deck and store in array of playercard or computercard (whichever is applicable).
+var drawHand = function (shuffledDeck, cardArray) {
+  var initialCard1 = shuffledDeck.pop();
+  var initialCard2 = shuffledDeck.pop();
+  cardArray.push(initialCard1);
+  cardArray.push(initialCard2);
```

Comment:
it's possible to do `cardArray.push(initialCard1, initialCard2)` nothing wrong with what you did, just wanted to let you know other possibilities.